pkg/workloadmeta/filter.go

```diff
@@ -50,31 +42,21 @@ func (f *Filter) MatchKind(k Kind) bool {
 // MatchSource returns true if the filter matches the passed sources. If the
 // filter is nil, or has no sources, it always matches.
 func (f *Filter) MatchSource(source Source) bool {
-	_, ok := f.SelectSources([]Source{source})
-
-	return ok
-}
-
-// SelectSources returns a subset of the passed sources that match the filter.
-func (f *Filter) SelectSources(sources []Source) ([]Source, bool) {
-	if f == nil || len(f.sources) == 0 {
-		return sources, true
+	if source == "" || f.Source() == "" {
```

Comment:
According to this (`source == ""`), when there's a filter that defines a source like Runtime, it will match events that don't have a source. It sounds a bit counter intuitive.

Reply:
Even though that shouldn't happen, I agree. Updated it to show intent more clearly.